element-hq/element-x-android
# Add warning message to 'mentions and keywords only' notification option

## Type of change

- [x] Other : add warning message.

## Content

- Adds `NotificationSettingsService. canHomeServerPushEncryptedEventsToDevice()` method to check if the home server can filter notifications for encrypted events containing mentions and keywords (the naming in Rust is a bit weird).
- Add `displayMentionsOnlyDisclaimer` parameter to all notification screens and components.
- Replace custom components for notification setting options with `ListItem`.

## Motivation and context

Closes #1794 .

## Screenshots / GIFs

Inside the PR.

## Tests

- Log into an account in matrix.org.
- Open the global notification settings screen, check this warning is displayed.
- Do the same for the notification settings of a room.

## Tested devices

- OS version(s): 11

## Checklist
